src/functions/keycloak/find-user/handler.ts

```diff
@@ -0,0 +1,85 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyHandler } from "aws-lambda";
+import { CreateKeycloakApiService, type KeycloakApiService } from "src/apis/keycloak-api-service";
+import { middyfy } from "src/libs/lambda";
+
+/**
+ * Response schema for lambda
+ */
+interface Response {
+  id: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  isActive: boolean;
+  severaGuid: string;
+  forecastId: number;
+}
+
+/**
+ * Gets user from keycloak
+ *
+ * @param api
+ * @param id
+ * @returns user information by Id
+ */
+const findUser = async (api: KeycloakApiService, id: string): Promise<Response[]> => {
+  const user = await api.findUser(id);
+  if (!user) {
+    throw new Error("Cannot find user from Api");
+}
+  return user;
+};
+
+/** 
+ * Lambda for finding user
+ * 
+ * @param _event event
+ * @returns user information as string
+ */
+
+const findUserHandler: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent) => {
+  const api = CreateKeycloakApiService();
+  const { queryStringParameters } = event;
+
+  // Ensure queryStringParameters and id are present
+  if (!queryStringParameters || !queryStringParameters.id) {
+    return {
+      headers:{
+        "Content-Type": "application/json"
+      },
+      statusCode: 400,
+      body: JSON.stringify({ error: "Missing or invalid path parameter: id" })
+    };
+  }
```

Comment:
How about having this within the try and handling the thrown error so that it is handled in the same catch you have below- you could pass the statusCode and error message?